larq/quantizers.py

```diff
@@ -141,9 +143,40 @@ def grad(dy):
     return _call(x)
 
 
+class BaseQuantizer(tf.keras.layers.Layer):
+    """Base class for defining quantizers with Larq metrics."""
+
+    def __init__(self, *args, metrics=None, **kwargs):
+        self._custom_metrics = (
+            metrics if metrics is not None else lq_metrics.get_training_metrics()
+        )
+
+        super().__init__(*args, **kwargs)
+
+    def build(self, input_shape):
+        super().build(input_shape)
+
+        if "flip_ratio" in self._custom_metrics:
+            self.flip_ratio = lq_metrics.FlipRatio(name=f"flip_ratio/{self.name}")
+
+    def call(self, inputs):
+        if hasattr(self, "flip_ratio"):
+
+            # If the flip ratio hasn't been built, first check if this is a weights
+            # quantizer; if not, remove the flip ratio.
+            if not self.flip_ratio.built and (
+                inputs.shape[0] is None or str(inputs.shape[0]) == "?"
```

Comment:
Could we do this shape check already in the build method?

In general what do you think about checking if `inputs` is a `QuantizedVariable` with `precision=1` here?